src/read_input_file.py

```diff
@@ -0,0 +1,35 @@
+"""File containing an input file reader for deb packages."""
+from typing import Set
+from pathlib import Path
+
+from src.package import PackageMetadata
+from src.get_package_version import get_package_version
+
+def _get_package_metadata(registry_path: Path, pinned_package: str) -> PackageMetadata:
+    if pinned_package.count(":") != 1 or pinned_package.count("=") > 1:
+        raise ValueError(
+            "Entry has unexpected format, expected format: name:arch=version or name:arch"
```

Comment:
```suggestion
            f"Entry {pinned_package!r} has unexpected format, expected 'name:arch=version' or 'name:arch'"
```
I think this would make debugging easier :)

Reply:
Agreed ! :)